microproxy/proxy.py

```diff
@@ -65,6 +66,9 @@ def handle_stream(self, stream, port):
                 yield process_result
         except gen.TimeoutError:
             stream.close()
+        except iostream.StreamClosedError:
+            logger.error("stream closed")
+            stream.close()
```

Comment:
dest_stream will be closed at Proxy Layer(Socks and Transparent)
Because dest_stream is not create from here, so it could make sure that dest stream is create or not and cannot close it.